jest.config.js

```diff
@@ -113,5 +113,7 @@ module.exports = {
     // https://github.com/storybookjs/storybook/pull/9795#issuecomment-587808019
     // either way, this can be removed in the future
     'react-syntax-highlighter/dist/esm/(.*)': 'react-syntax-highlighter/dist/cjs/$1',
+    '@carbon/icons-react/es/(.*)': '@carbon/icons-react/lib/$1',
```

Comment:
Some commenting to notate the history of what's/why happening here
```suggestion
    // rewrite carbon-components(-react) es imports to lib/cjs imports because jest doesn't support es modules
    // https://github.com/facebook/jest/issues/4842
    '@carbon/icons-react/es/(.*)': '@carbon/icons-react/lib/$1',
```